app/routes/goal_routes.py

```diff
@@ -1 +1,119 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, Response
+from flask import request, abort, make_response
+from ..models.goal import Goal
+from ..models.task import Task
+from .route_utilities import validate_model, create_model, get_models_with_filters
+from ..db import db
+from sqlalchemy import asc, desc
+from flask import jsonify
+
+
+bp = Blueprint("goals_bp", __name__, url_prefix = "/goals")
+
+@bp.post("")
+def create_new_goal():
+    request_body = request.get_json()
+    if (
+        not request_body
+        or "title" not in request_body
+    ):
+        abort(make_response({"details": "Invalid data"}, 400))
+    new_goal = Goal.from_dict(request_body)
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {
+            "goal":{
+            "id" : new_goal.id,
+            "title": new_goal.title
+            }
+        }, 201
+
+@bp.post("/<id>/tasks")
+def create_tasks_with_goal_id(id):
+    goal = validate_model(Goal, id)
+    request_body = request.get_json()
+    task_ids = request_body.get("task_ids", [])
+    if not isinstance(task_ids, list):
+        abort(make_response({"details": "Invalid data"}, 400))
+    tasks = db.session.query(Task).filter(Task.id.in_(task_ids)).all()
```

Comment:
Instead of directly querying the DB, I'd like to see the use of `validate_model` since you've already created this helper function which has error handling logic too.
```suggestion
    tasks = [validate_model(task_id) for task_id in task_ids]
```